src/Features/LanguageServer/Protocol/Handler/RequestExecutionQueue.cs

```diff
@@ -178,6 +178,11 @@ public void Shutdown()
                 _requestTelemetryLogger,
                 combinedCancellationToken);
 
+            // Run a continuation to ensure the cts is disposed of.
+            // We pass in the queue's cancellation token to the continuation as we should dispose
+            // of the source even if the client cancels the request.
+            resultTask.ContinueWith((_) => combinedTokenSource.Dispose(), _cancelSource.Token, TaskContinuationOptions.None, TaskScheduler.Default);
```

Comment:
probably don't want this disposable to itself be cancellable right?  also, probably good to RunSynchronously if we can to keep things lighter on the threadpool.

Reply:
yeah that is true actually - even if the queue stops we probably want to dispose of all the sources.